proxy/src/main.rs

```diff
@@ -116,11 +116,10 @@ async fn handle_signals() -> anyhow::Result<()> {
                 warn!("received SIGINT, exiting immediately");
                 bail!("interrupted");
             }
-            // TODO: Don't accept new proxy connections.
-            // TODO: Shut down once all exisiting connections have been closed.
             _ = terminate.recv() => {
-                warn!("received SIGTERM, exiting immediately");
-                bail!("terminated");
+                warn!("received SIGTERM, shutting down once all existing connections have closed");
```

Comment:
Because the current behavior is to shut down immediately on `SIGTERM` and it's also the default signal for `kill`, I think it would potentially cause less surprises if we kept the current behavior and made the graceful shutdown require something else (QUIT, INT).

Don't think there's really any global standard which one exactly as it's a bit different with for example nginx, haproxy, envoy, postgresql and so on. What do you think?

Reply:
https://www.postgresql.org/docs/current/server-shutdown.html

It's a kind of inverted logic in my mind. Sigterm is used for "smart" shutdown.

Reply:
I disagree with Lassi's proposal. I think we should stick with [what postgres does](https://www.postgresql.org/docs/current/server-shutdown.html).